src/metabase/api/pulse.clj

```diff
@@ -135,11 +136,15 @@
                  ;; if we have Slack enabled build a dynamic list of channels/users
                  :else
                  (try
-                   (let [slack-channels (for [channel (slack/conversations-list)]
-                                          (str \# (:name channel)))
-                         slack-users    (for [user (slack/users-list)]
-                                          (str \@ (:name user)))]
-                     (assoc-in chan-types [:slack :fields 0 :options] (concat slack-channels slack-users)))
+                   (let [[conversations users] (map deref [(future (slack/conversations-list-timeout))
+                                                           (future (slack/users-list-timeout))])
```

Comment:
this way we don't need to wait for the conversations list to finish to start getting the users list.

These *-timeout functions return a map like `{:result [...] :timeout true}` if the timeout was hit and we need to return the cached version.